projects/plugins/wpcomsh/performance-profiler/performance-profiler.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Adds the generated performance report url for each page and also in v1 site settings payload for homepage.
+ *
+ * @package performance-profiler
+ */
+
+/**
+ * Adds the generated performance report url for each page.
+ */
+function wpcom_performance_url_rest_field() {
+	register_rest_field(
+		'page',
+		'wpcom_performance_report_url',
+		array(
+			'get_callback'    => 'wpcom_performance_report_url_get_value',
+			'update_callback' => 'wpcom_performance_report_url_update_value',
+			'schema'          => array(
+				'type'        => 'string',
+				'description' => __( 'The performance report URL for this page.', 'wpcomsh' ),
+				'default'     => '',
+			),
+		)
+	);
+}
+
+/**
+ * Filter the performance report URL field.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Post          $post     The post object.
+ * @param WP_REST_Request  $request  The request object.
+ * @return WP_REST_Response
+ */
+function filter_wpcom_performance_report_url( $response, $post, $request ) {
+	$fields = $request->get_param( '_fields' );
+
+	if ( is_string( $fields ) ) {
+		$fields = explode( ',', $fields );
+	}
+
+	if ( ! is_array( $fields ) ) {
+		$fields = array();
+	}
+
+	if ( empty( $fields ) || ! in_array( 'wpcom_performance_report_url', $fields, true ) ) {
+		unset( $response->data['wpcom_performance_report_url'] );
+	}
+
+	return $response;
+}
+
+add_filter( 'rest_prepare_page', 'filter_wpcom_performance_report_url', 10, 3 );
+
+/**
+ * Get the performance report URL callback.
+ *
+ * @param array $object The object being requested.
+ * @return string|null The performance report URL.
+ */
+function wpcom_performance_report_url_get_value( $object ) {
+	return get_post_meta( $object['id'], '_wpcom_performance_report_url', true );
+}
+
+/**
+ * Update callback for the performance report URL.
+ *
+ * @param  mixed  $value The new value of the field.
+ * @param  object $object The object being updated.
+ * @return bool|int|null
+ */
+function wpcom_performance_report_url_update_value( $value, $object ) {
+	if ( ! $value || ! is_string( $value ) ) {
+		return;
+	}
+	return update_post_meta( $object->ID, '_wpcom_performance_report_url', sanitize_text_field( $value ) );
```

Comment:
Would `sanitize_url` be the better option here? Unless you're planning on only storing the hash.

```suggestion
	return update_post_meta( $object->ID, '_wpcom_performance_report_url', sanitize_url( $value ) );
```